pkg/api/v1/routes/handlers.go

```diff
@@ -227,20 +240,26 @@ func (r *Routes) serverEnroll(c *gin.Context) (int, *v1types.ServerResponse) {
 	// Delete params related to add servers.
 	var inventoryArgs rctypes.InventoryTaskParameters
 	if err = json.Unmarshal(conditionCreate.Parameters, &inventoryArgs); err != nil {
+		shouldCleanup = true
 		return http.StatusBadRequest, &v1types.ServerResponse{
 			Message: "invalid inventory params: " + err.Error(),
 		}
 	}
 	inventoryParams, err := json.Marshal(inventoryArgs)
 	if err != nil {
+		shouldCleanup = true
 		return http.StatusBadRequest, &v1types.ServerResponse{
 			Message: err.Error(),
 		}
 	}
 	conditionCreate.Parameters = inventoryParams
 	newCondition := conditionCreate.NewCondition(rctypes.Inventory)
 
-	return r.conditionCreate(otelCtx, newCondition, serverID, params.Facility)
+	st, resp := r.conditionCreate(otelCtx, newCondition, serverID, params.Facility)
+	if st != http.StatusOK {
+		shouldCleanup = true
```

Comment:
By eliminating the marshaling stuff, we can call `cleanup` here directly instead of `deferring` it.